core/chain_manager.go

```diff
@@ -335,23 +319,23 @@ func (self *ChainManager) ExportN(w io.Writer, first uint64, last uint64) error
 // insert injects a block into the current chain block chain. Note, this function
 // assumes that the `mu` mutex is held!
 func (bc *ChainManager) insert(block *types.Block) {
-	err := WriteHead(bc.chainDb, block)
-	if err != nil {
-		glog.Fatal("db write fail:", err)
+	// Insert the block as the new head of the chain
```

Comment:
OK, then please either update the comment or move it down to WriteHeadBlockHash. It sort-of reads like WriteCanonicalHash is having something todo with head of chain.